### ISSUE TITLE
Carousel prev/next slide incorrect after double clicking page control

### ISSUE THREAD

**Issue by Skawt2k** [angular-ui/bootstrap#5967]
### Bug description:

When double clicking the previous/next control or clicking the previous/next control again before the current page transition has completed causes the controls and looping behaviour of the carousel to be incorrect/inconsistent. 
### Steps to reproduce

1) Visit http://angular-ui.github.io/bootstrap/ . The active slide is 0.

2) Double click the carousel next control. The active slide is now 1 as expected.

3) Click the previous control. The active slide now becomes 2, the carousel should actually transition to the previous slide instead of the next slide. 

4) Move the pointer away from the carousel (out). The carousel will move to the next slide, this is not expected.

Note: This also happens after quick swipe successions with touch enabled. 
### Link that reproduces the issue:

This is an issue on the current angular-ui bootstrap demo site:
http://angular-ui.github.io/bootstrap/
### Version of Angular, UIBS, and Bootstrap

Angular: 1.5.3

UIBS: 1.3.3

Bootstrap: 3.6.6

**Comment by icfantv** [comment]
@Skawt2k we're working on removing `replace: true` from our code to make `ng-class` directives work (as well as resolve other issues).  as such the code has changed.  Can you please try building against master and test this again?  Thanks.

**Comment by wesleycho** [comment]
No need for that - this sounds like a potential issue with $animate not resolving correctly, it may be an Angular ngAnimate bug, but I need to investigate.

**Comment by TechCraver1** [comment]
@wesleycho Add label needs: info.

**Comment by icfantv** [comment]
@wesleycho, ok, sounds good.

**Comment by stuartraines** [comment]
@wesleycho - Have you managed to investigate this issue at all? Is there a workaround that you know of?

**Comment by tariqporter** [comment]
@stuartraines  Yes, for now you can go to line approx 541 and comment out:

```
else if (nextSlide && nextSlide.slide.index !== currentIndex && $scope.$currentTransition) {
    bufferedTransitions.push(slides[nextIndex]);
}
```

**Comment by Skawt2k** [comment]
@tariqporter - Many thanks for the workaround. I can confirm that commenting out the lines specified above resolves the issue. Thx.

**Comment by stuartraines** [comment]
@tariqporter - Thanks very much for taking the time to look into this and for the fix

**Comment by kive323** [comment]
I have comment out that part, but the issue still happens, any other solution I can fix that.

**Comment by kive323** [comment]
@tariqporter - I have comment out that part, but the issue still happens, any other solution I can fix that.

**Comment by tariqporter** [comment]
@kive323 For any version of angular-ui, this solution has worked for me. I assume that code is meant to be buffering clicks that are too quick. For my purposes, this is not needed, so removing buffering is fine. Test with a new version of angular-ui

**Comment by umeshdeveloper** [comment]
In  which file the code exists, I made a search in my ui-bootstrap-tpls-2.1.4.min.js, angular.min.js. couldn't find the above code to comment it out.

**Comment by tariqporter** [comment]
I have created a pull request to properly fix this issue, and not use the hack I provided above

[pull request #6367](https://github.com/angular-ui/bootstrap/pull/6367)

**Comment by swapnilbhalgat** [comment]
I'm still facing this issue, when I click on next button multiple times and click on prev button and click back on next button, it goes to previous slide. Sometimes, when I keeo clicking on next button it will go to prev slide. Can you please tell me if this issue is fixed or is there a workaround (without using the hack)?

**Comment by tariqporter** [comment]
@swapnilbhalgat  As far as I know, the last update that I pushed fixed this issue in 2.5.0. I haven't seen it since
